File: src/protocol/AutomaticBlindsBuilder.cpp
#include "AutomaticBlindsBuilder.h"
#include "AutomaticBlindsCommand.h"

using namespace std;
using namespace communication;

AutomaticBlindsBuilder::AutomaticBlindsBuilder()
{}

AutomaticBlindsBuilder::~AutomaticBlindsBuilder()
{}

std::unique_ptr<Command> AutomaticBlindsBuilder::create(const std::vector<uint8_t> &commandInBytes)
{
    auto command = make_unique<AutomaticBlindsCommand>();

    return move(command);
}

File: mobile_client/app/src/main/java/protocol/AutomaticBlindsBuilder.java
package protocol;

import java.io.Closeable;
import java.util.ArrayList;

public final class AutomaticBlindsBuilder implements CommandBuilder, Closeable
	{
		public AutomaticBlindsBuilder()
		{
		}
		public final void close()
		{
		}

		@Override
		public Command create(ArrayList<Integer> commandInBytes)
		{
			Command command  = new AutomaticBlindsCommand();
			return command;
		}
	}